external-crates/move/crates/move-package-alt/src/graph/package_info.rs

```diff
@@ -161,7 +175,24 @@ impl<'graph, F: MoveFlavor> PackageInfo<'graph, F> {
     /// addresses as well
     pub fn named_addresses(&self) -> PackageResult<BTreeMap<PackageName, NamedAddress>> {
         if self.package().is_legacy() {
-            return self.legacy_named_addresses();
+            // Build up the context from the root, so we can pass addresses down.
+            // This is empty if the package is the root package.
+            let root_assignments = if let Some(incoming) = self
+                .graph
+                .inner
+                .edges_directed(self.node, Direction::Incoming)
+                .next()
+            {
+                let parent = PackageInfo {
+                    graph: self.graph,
+                    node: incoming.source(),
+                };
+                parent.named_addresses()?
```

Comment:
this might do the wrong thing in the legacy -> modern -> legacy case, but we think it's ok.